supplier/tests/test_supplier_page.py

```diff
@@ -39,3 +91,66 @@ def test_supplier_check_false(non_supplier_mock):
 def test_supplier_check_true(saved_supplier0):
     assert saved_supplier0 in Supplier.objects.all()
     assert get_supplier(saved_supplier0) == saved_supplier0
+
+
+@pytest.mark.django_db()
+def test_receive_product_return_supplier_product(request_searched_product_mock,
+                                                 saved_supplier0,
+                                                 saved_supplier_product0,
+                                                 saved_supplier_product1,
+                                                 searched_product_string):
+    assert saved_supplier_product0.user_name == saved_supplier0
+    assert saved_supplier_product1.user_name == saved_supplier0
+    request_searched_product_mock.GET[searched_product_string] = saved_supplier_product0.qr_code.product_name
+    assert receive_product(request_searched_product_mock, saved_supplier0).first() == saved_supplier_product0
+
+
+@pytest.mark.django_db()
+def test_receive_product_return_none(request_searched_product_mock,
```

Comment:
Your code have two branches in which `None` is returned (else of the first `if` and else of the second one), but you only have one test case here. You need to have them both covered.